inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala

```diff
@@ -93,6 +112,32 @@ final case class PublicKey(value : String) extends HashcodeCaching { assert(!isE
 
 }
 
+final case class Certificate(value : String) extends SecurityToken with HashcodeCaching { assert(!isEmpty(value))
+
+  // Value of the key may be stored (with old fusion inventory version) as one line and without rsa header and footer, we should add them if missing and format the key
+  val key = {
+    if (value.startsWith("-----BEGIN CERTIFICATE-----")) {
+      value
+    } else {
+      s"""-----BEGIN CERTIFICATE-----
+      |${value.grouped(80).mkString("\n")}
```

Comment:
Yes, that the format of the certificate we generate is PEM which is  "-----BEGIN CERTIFICATE-----" and "-----END CERTIFICATE-----" for a x509 certificate 

(https://en.wikipedia.org/wiki/X.509#Certificate_filename_extensions and read from Boundy Castle PEMParser )

or a SO answer here : https://stackoverflow.com/questions/3313020/write-x509-certificate-into-pem-formatted-string-in-java